src/docfx/build/toc/BuildTableOfContents.cs

```diff
@@ -53,6 +53,7 @@ internal static class BuildTableOfContents
                 Path = outputPath,
                 Locale = file.Docset.Locale,
                 Monikers = model.Metadata.Monikers,
+                Group = HashUtility.GetMd5HashShort(model.Metadata.Monikers),
```

Comment:
calculating hash is slow, `MonikerGroup` can simply be an increasing number starting from 1 and assigned during `PublishModelBuilder.Build`, this can also produce deterministic result.

Reply:
that will cause different monikers in the different repo to produce the same group name, which may be not acceptable for DHS. @fenxu any suggestion?
For perf concern, we can add a cache layer to avoid duplicate calculation.

Reply:
1. Keep result stable by using Hash.
2. Improve perf by cache

---
In reply to: [285826404](https://github.com/dotnet/docfx/pull/4589#discussion_r285826404) [](ancestors = 285826404)